R/pmcmc_state.R

```diff
@@ -0,0 +1,191 @@
+pmcmc_state <- R6::R6Class(
```

Comment:
yes, sorry - if we were not over the Christmas break I'd have done this in 2 PRs so that's clear. This is just a code move with some book-keeping moved.

* Move code: https://github.com/mrc-ide/mcstate/compare/master...6aaafd4a025e32e065eb5875b311b4a74576b70d
* Implement parallel control: https://github.com/mrc-ide/mcstate/compare/6aaafd4a025e32e065eb5875b311b4a74576b70d...i88-parallel